lib/logstash/filters/geoip.rb

```diff
@@ -124,22 +124,24 @@ def register
 
   public
   def filter(event)
-    
+
     geo_data = nil
 
     geo_data = get_geo_data(event)
 
-    return if geo_data.nil? || !geo_data.respond_to?(:to_hash)
-
-    apply_geodata(geo_data, event)
+    return unless geo_data.respond_to?(:to_hash)
 
-    filter_matched(event)
+    event[@target] = {} if event[@target].nil?
+    geo_data_hash = geo_data.to_hash
+    # don't do anything more if the lookup result is empty
+    if !geo_data_hash.empty?
```

Comment:
an alternative to line 135 + checking if it's empty here you can just check if geo_data != {}